client/src/controllers/TeleportController.ts

```diff
@@ -88,22 +88,19 @@ export class TeleportController extends Controller<HTMLTemplateElement> {
   }
 
   /**
-   * Resolve a valid HTMLElement from the controlled element's children.
+   * Returns a fresh copy of the DocumentFragment from the controlled element.
    */
-  get templateElement() {
-    const templateElement =
-      this.element.content.firstElementChild?.cloneNode(true);
-
-    if (!(templateElement instanceof HTMLElement)) {
-      throw new Error('Invalid template content.');
-    }
+  get templateFragment() {
+    const templateFragment = this.element.content.cloneNode(
+      true,
+    ) as DocumentFragment;
```

Comment:
A bit verbose but it may be clearer for later if we call this out explicitly.

```suggestion
    const content = this.element.content;
    // https://developer.mozilla.org/en-US/docs/Web/API/Node/cloneNode (returns the same type)
    // https://github.com/microsoft/TypeScript/issues/283 (TypeScript will return as Node, incorrectly)
    const templateFragment = content.cloneNode(true) as typeof content;
```